app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCapturePresenterImpl.java

```diff
@@ -295,14 +297,23 @@ public void init() {
                                                         }
                                                     }
 
+                                                    if (fieldMap.get("") != null) {
+                                                        finalFieldList.addAll(fieldMap.get(""));
+                                                    }
+
                                                     return Pair.create(eventSectionModels, finalFieldList);
                                                 })))
                         .subscribeOn(schedulerProvider.io())
                         .observeOn(schedulerProvider.ui())
                         .subscribe(sectionsAndFields -> {
-                                    if (assignedValueChanged) {
+                                    if (assignedValueChanged && errors.isEmpty() && calculationLoop < 5) {
+                                        calculationLoop++;
                                         nextCalculation(true);
                                     } else {
+                                        if (calculationLoop == 5) {
+                                            view.showLoopWarning();
```

Comment:
can you use a constant? something like ; MAX_RETRY

Reply:
On my way! :)